src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -94,12 +96,117 @@ void EventPipe::Initialize()
 #endif
     }
 
-
     {
         CrstHolder _crst(GetLock());
         if (tracingInitialized)
             s_state = EventPipeState::Initialized;
     }
+#ifdef FEATURE_EVENTPIPE_STARTUP
+    EnableViaEnvironmentVariables();
+#endif // FEATURE_EVENTPIPE_STARTUP
+}
+
+//
+// If EventPipe environment variables are specified, parse them and start a session
+// 
+void EventPipe::EnableViaEnvironmentVariables()
+{
+    STANDARD_VM_CONTRACT;
+    if (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EnableEventPipe) != 0)
+    {
+        CLRConfigStringHolder eventpipeConfig(CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EventPipeConfig));
+        CLRConfigStringHolder configOutputPath(CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EventPipeOutputPath));
+        uint32_t eventpipeCircularBufferMB = CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EventPipeCircularMB);
+        LPCWSTR outputPath = nullptr;
+
+        if (configOutputPath == NULL)
+        {
+            outputPath = W("trace.nettrace");
+        }
+        else
+        {
+            outputPath = configOutputPath;
+        }
+        auto configuration = XplatEventLoggerConfiguration();
+        LPWSTR configToParse = eventpipeConfig;
+        int providerCnt = 0;
+
+        // Create EventPipeProviderConfiguration and start tracing.
+        EventPipeProviderConfiguration* pProviders = nullptr;
+
+        // If COMPlus_EnableEventPipe is set to 1 but no configuration was specified, enable EventPipe session
+        // with the default provider configurations.
+        if (configToParse == nullptr || *configToParse == L'\0')
+        {
+            providerCnt = 2;
+            pProviders = new EventPipeProviderConfiguration[providerCnt];
+            pProviders[0] = EventPipeProviderConfiguration(W("Microsoft-Windows-DotNETRuntime"), 0x4c14fccbd, 5, nullptr);
+            pProviders[1] = EventPipeProviderConfiguration(W("Microsoft-Windows-DotNETRuntimePrivate"), 0x4002000b, 5, nullptr);
+        }
+        else
+        {
+            // Count how many providers there are to parse
+            static WCHAR comma = W(',');
+            while (*configToParse != '\0')
+            {
+                providerCnt += 1;
+                auto end = wcschr(configToParse, comma);
+                if (end == nullptr)
+                {
+                    break;
+                }
+                configToParse = end + 1;
+            }
+            configToParse = eventpipeConfig;
+            pProviders = new EventPipeProviderConfiguration[providerCnt];
+            int i = 0;
+            while (*configToParse != '\0')
+            {
+                auto end = wcschr(configToParse, comma);
+                configuration.Parse(configToParse);
+
+                // SampleProfiler can't be enabled on startup yet.
+                if (wcscmp(W("Microsoft-DotNETCore-SampleProfiler"), configuration.GetProviderName()) == 0)
```

Comment:
If `configuration.IsValid()` is `false`, `configuration.GetProviderName()` returns `nullptr`.  This `wcscmp()` will crash.